.github/workflows/build-plugin-archive.yml

```
@@ -0,0 +1,183 @@
+        type: string
+      PACKAGE_MANAGER:
+          description: Package manager with which the dependencies should be installed (`npm` or `yarn`).
+          default: 'yarn'
+          required: false
+          type: string
+    secrets:
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+      NPM_REGISTRY_TOKEN:
+        description: Authentication for the private npm registry.
+        required: false
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  create-plugin-archive:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    env:
+      # Disables symlinking of local path repositories.
+      # During development, symlinking is preferable.
+      # In resulting builds, you will likely want to ship the actual install location and remove the source directory.
+      COMPOSER_MIRROR_PATH_REPOS: 1
+      COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+    outputs:
+      artifact: ${{ steps.set-artifact-name.outputs.artifact }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup plugin base name
+        env:
+          ARCHIVE_NAME: ${{ inputs.ARCHIVE_NAME }}
+        id: plugin-data
+        run: |
+          if [ ! -z "$ARCHIVE_NAME" ]; then
+              echo "archive-name=$ARCHIVE_NAME" >> $GITHUB_OUTPUT
+          else
+              echo "archive-name=${{ github.event.repository.name }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up custom environment variables
+        env:
+          ENV_VARS: ${{ secrets.ENV_VARS }}
+        if: ${{ env.ENV_VARS }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: wp-cli
+
+      - name: Install Composer dependencies with dev dependencies
+        # ⬑ The idea is to ensure we have non-production tools like composer-assets-compiler and translation downloader available
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: --prefer-dist
+
+      - name: Check optional Composer build tools
+        id: composer-tools
+        run: |
+          hasDep(){
+            composer show -i -N -D --strict "${1}" >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          echo "assets-compiler=$( hasDep inpsyde/composer-assets-compiler )" >> $GITHUB_OUTPUT
+          echo "translation-downloader=$( hasDep inpsyde/wp-translation-downloader )" >> $GITHUB_OUTPUT
+
+      - name: Run Composer Assets Compiler
```

The official name is "Composer Asset Compiler", without the "s".
```suggestion
      - name: Run Composer Asset Compiler
```